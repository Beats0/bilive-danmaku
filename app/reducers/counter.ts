import { Action } from 'redux';
import { INCREMENT_COUNTER, DECREMENT_COUNTER } from '../actions/counter';
import { counterStateType } from './types';

const initState: counterStateType = {
  count1: 0,
  count2: 0
};

export default function counter(state = initState, action: Action<string>) {
  switch (action.type) {
    case INCREMENT_COUNTER:
      return { ...state, count1: state.count1 + 1 };
    case DECREMENT_COUNTER:
      return { ...state, count1: state.count1 - 1 };
    default:
      return state;
  }
}
