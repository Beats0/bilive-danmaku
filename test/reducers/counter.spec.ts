import counter from '../../app/reducers/counter';
import {
  INCREMENT_COUNTER,
  DECREMENT_COUNTER
} from '../../app/actions/counter';

describe('reducers', () => {
  describe('counter', () => {
    it('should handle initial state', () => {
      expect(counter({ count1: 0, count2: 0 }, {})).toMatchSnapshot();
    });

    it('should handle INCREMENT_COUNTER', () => {
      expect(
        counter({ count1: 0, count2: 0 }, { type: INCREMENT_COUNTER })
      ).toMatchSnapshot();
    });

    it('should handle DECREMENT_COUNTER', () => {
      expect(
        counter({ count1: 1, count2: 0 }, { type: DECREMENT_COUNTER })
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        counter({ count1: 0, count2: 0 }, { type: 'unknown' })
      ).toMatchSnapshot();
    });
  });
});
