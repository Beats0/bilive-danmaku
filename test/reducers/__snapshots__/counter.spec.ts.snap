// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle DECREMENT_COUNTER 1`] = `
Object {
  "count1": 0,
  "count2": 0,
}
`;

exports[`reducers counter should handle INCREMENT_COUNTER 1`] = `
Object {
  "count1": 1,
  "count2": 0,
}
`;

exports[`reducers counter should handle initial state 1`] = `
Object {
  "count1": 0,
  "count2": 0,
}
`;

exports[`reducers counter should handle unknown action type 1`] = `
Object {
  "count1": 0,
  "count2": 0,
}
`;
